### PR TITLE
changes to h5 loaders to have res2rad

### PR DESCRIPTION
**Pull request by rrstofer**

### CODE REVIEW

**Comment by rrstofer** [comment]
It also has an additional L2 argument for PNG loaders so if the loss is L2, it can run normally!

File: loaders.py [hunk 1]
```diff
@@ -55,7 +58,7 @@ def __init__(self, pngdir=None, propfile=None, quad="A", start=None, stop=None,
             self.prop.loc[:,"reso"] = 1/self.prop.reso
 
         if convert_res:
-            self.prop.loc[:, "reso"] = self._convert_res2rad(self.prop.reso)
+            self.prop.loc[:, "reso"] = 1/self._convert_res2rad(self.prop.reso)
```
Comment (dermen) [inline loaders.py]: I think we need all 4 options: either use res, one_over_res, rad, one_over rad. You can then remove the assert on line 33 (now we will allow invert_res and convert_res to both be True). Then switch the order, such that inversion happens after conversion..

File: loaders.py [hunk 2]
```diff
@@ -117,13 +120,17 @@ def __getitem__(self, i):
         # Apply image transform here
         if self.transform:
             img_dat = self.transform(img_dat)
+        if self.L2:
+            img_dat = img_dat.float()
```
Comment (dermen) [inline loaders.py:121]: remind me again what was happening here ? Why was this necessary?

Comment (dermen) [reply loaders.py:121]: Also, I would change the variable `self.L2` to be something more descriptive, like `self.convert_to_float`..

File: loaders.py [hunk 3]
```diff
@@ -173,12 +181,30 @@ def __init__(self, h5name, dev=None, labels="labels", images="images",
         self.labels = None  # hdf5 dataset
         self.geom = None  # hdf5 dataset
 
+    def _convert_res2rad(self, res):
```
Comment (dermen) [inline loaders.py]: So if you look in `sims/main.py`, where the labels are created, you will notice a resolution -> radius conversion occurs `radius = reso2radius(params["reso"], DET, BEAM)`,  and this radius is stored in the hdf5 file. 

So one simply needs to create a dataset to point to it (see `scripts/expand_labels.py`.. Hence, I dont think we need this here.

**Review by dermen — CHANGES_REQUESTED** [review]
Looks good to support all 4 options in the PngDset class (invert /convert).

But regarding the h5 loader having a convert script, I dont think we need it now. You could though update the `sims/main.py` to store a one_over_radius in the labels , and then update `scripts/expand_labels.py` accordingly.

File: loaders.py (hunk 2 above)
Comment (Sir-Teo) [reply loaders.py:121]: I changed this on my end also, cos for MESLoss, it only support the float datatype

**Comment by dermen** [comment]
Added some updates. Basically, going forward lets simply use the net.py script. First, one must convert the Png data to HDF5 using the new script `resmos_to_hdf5.py`

```
python resmos_to_hdf5.py  $DD/png /data/dermen/num_reso_mos_B_icy1_icy2_cell_SGnum_pdbid_stolid.txt  $DD/resmos2.hdf5 --overwrite
```

Then, one can train on rad, one_over_rad, res, one_over_res, using e.g.:

```
python net.py 100 resmos2.hdf5 test_resmos2_h5 --lr 1e-4 --bs 16   --saveFreq 5 --arch res50 --labelName one_over_rad --loss L2 --testRange 0 500 --trainRange 500 2500 --momentum 0.9 
```

**Review by dermen — APPROVED** [review]

**Pull request merged**